### ISSUE TITLE
Parent of a null child node in trie is masked when parent line not present

### ISSUE THREAD

**Issue by lmmx** [lmmx/wotd#1]
While establishing input validation for 

```sh
grep -v "\- \[x\]" ukp_manifest.md | grep "\[\].*\[\]" | cut -d\` -f2 | tail -6 | sed '/variants$/d' | python trie_walk.py -
```
⇣
```STDOUT
.[] .tweet .extended_entities .media[] .video_info .aspect_ratio
                                                   .duration_millis
                                                               .bitrate
                                                               .content_type
                                                               .url
```

I noticed there's a bug here. The output should be:

```
.[] .tweet .extended_entities .media[] .video_info .aspect_ratio
                                                   .duration_millis
                                                   .variants[] .bitrate
                                                               .content_type
                                                               .url
```

The difference between this (the expected output) and the invalid masked trie output (with the first line ending with `.video_info`) is the difference between valid and invalid output.

- As far as I can tell so far, valid input will have no 'null children', as any descendants must (by the definition of mutually exclusive descendants and ancestors) exclude their descendants from a list of filter paths.
- In other words, detecting paths which have these 'null children' will be sufficient to validate the masked trie list as input to `build_del_call.py`.

I think this should go in a separate file as `trie_walk.py` has hitherto been able to assume parent nodes could be present alongside descendants, but when used as a filter then this assumption becomes a prohibition.

This is called iterator masking in [section 4 of AUDIT.md](https://github.com/lmmx/wotd/blob/master/AUDIT.md#4-subtle-but-important-iterator-masking).
